File: python/src/collatz_conjecture.py
def steps(number):
    if number < 1:
        raise ValueError("Only positive integers are allowed")
    steps = 0
    current = number
    while current != 1:
        if current % 2 == 0:
            current /= 2
        else:
            current = 3 * current + 1
        steps +=1
    return steps

File: typescript/src/collatz-conjecture.ts
export function steps(count: number): number {
  if (count<=0 || !Number.isInteger(count)) {
    throw new Error("Only positive integers are allowed")
  }
  
  if (count == 1) {
    return 0;
  } else {
    return 1 + steps(count % 2 == 0 ? count / 2 : 3 * count + 1)
  }
}
